described embeddings (#5)
SpecialeBajs/speciale-rapport: article/Method/embeddings.tex

\subsection{Brief Review of NGCF, LightGCN and GCF}
Neural Graph Collaborative Filtering (NGCF) was published in 2019, and was a state of the art method, that utilized graph convolutional networks (GCN) for collaborative filtering \cite{NGCF_2019}.

> The original use for GCN was proposed on node classification where each node has rich attributes, but for collaborative filtering the nodes only contains node IDs \cite{lightgcn,kipf2017semisupervised,NGCF_2019}.

Reviewer: ```suggestion
The original use for GCN was proposed on node classification where each node has rich attributes, but for collaborative filtering, the nodes only contain node IDs \cite{lightgcn,kipf2017semisupervised,NGCF_2019}.
```

useriøst

> This is something that LightGCN took advantage of and simplified NGCF by removing the non linear activation function and the feature transformation, which showed promising results in their experiments.

Reviewer: ```suggestion
This is something that LightGCN took advantage of and simplified NGCF by removing the nonlinear activation function and the feature transformation, which showed promising results in their experiments.
```

> Later Meng Liu et. al then created a method called BiTGCF which utilized GCN and transfer learning between two different datasets \cite{BiTGCF}.

Reviewer 2: ```suggestion
Later Meng Liu et. al then created a method called BiTGCF which utilized GCN and transfer learning between two domains \cite{BiTGCF}.
```

In this paper we only concentrate on single domain methods, and therefore will only focus on GCF of these two methods.

\subsubsection{Graph Convolutional Networks}
GCN for collaborative filtering can generally be abstracted as \cite{BiTGCF}
\begin{equation}
    e_u^{(k+1)} = AGG(e_u^{(k)},e_i^{(k)} : i \in \mathcal{N}_u),
\end{equation}
where $AGG(\cdot)$ is an aggregation function, and $e_i^{(k)}$ and $e_u^{(k)}$ denotes the embeddings for user $u$ and item $i$ after $k$ propagation layers.
The embedding propagation for items can be obtained similarly as the embeddings with users by replacing $u$ with $i$ and vice versa.
$\mathcal{N}_u$ and $\mathcal{N}_i$ denotes the one hop neighbors from user $u$ and item $i$ respectively \cite{lightgcn,BiTGCF,NGCF_2019}.\\
Following this section we will describe the embedding propagation for users and the final layer combination for NGCF, LightGCN and GCF.

\subsubsection{Embedding Propagation in NGCF:}

> The embedding propagation for NGCF is defined as,

Reviewer 2: ```suggestion
The embedding propagation in NGCF is defined as,
```

\begin{equation}
    \mathbf{e}_{u}^{(k+1)} = \mbox{LeakyReLU}(\mathbf{m}^{(k)}_{u \leftarrow u} + \sum^{}_{i \in \mathcal{N}_u} \mathbf{m}^{(k)}_{u \leftarrow i}),
    \label{eq:ngcf-embedding-propagation}
\end{equation}

> where LeakyReLU is the nonlinear activation function and $\mathbf{m}^{(k)}_{u \leftarrow i}$ is the message construction from users u to item i, and $\mathbf{m}^{(k)}_{u \leftarrow u}$ is the self connection.

Reviewer 2: ```suggestion
where LeakyReLU is the nonlinear activation function, $\mathbf{m}^{(k)}_{u \leftarrow i}$ is the message construction from users u to item i, and $\mathbf{m}^{(k)}_{u \leftarrow u}$ is the self connection.
```

\end{equation}
As can be seen, LightGCN simplified NGCF by removing the activation function, transformation matrices, the self loop and the inner product between the user and item embeddings.
Additionally LightGCN changes the layer combination from concatenation to weighted summation as shown here:
\begin{equation}
    \mathbf{e}_u = \sum_{k=0}^{K} \alpha_k \mathbf{e}_u^{(k)},
    \label{eq:lightgcn-sum}
\end{equation}
where $K$ is the total number of layers and $\alpha_k$ is a hyper parameter used to denote the importance of the $k$-th embedding \cite{lightgcn}.
$\alpha_k$ is set to $1 /(K + 1)$ for simplicity.

\subsubsection{Embedding Propagation in GCF.}
The embedding propagation in GCF is defined as \cite{BiTGCF},
\begin{equation}
    \mathbf{e}_{u}^{(k+1)} = \mathbf{e}_{u}^{(k)} + \sum^{}_{i \in \mathcal{N}_u}  \frac{1}{\sqrt{|\mathcal{N}_u||\mathcal{N}_i|}}\left( \mathbf{e}_i^{(k)} + \mathbf{e}_i^{(k)} \odot \mathbf{e}_u^{(k)} \right)
\end{equation}
As can be seen GCF adds the self connections and the inner product between the users and items compared to LightGCN.

> The purpose of the inner product is that the more the user and item nodes has in common the greater the value will be passed to the center node \cite{BiTGCF}.

Reviewer 2: ```suggestion
The purpose of the inner product is that the more the user and item nodes have in common the greater the value will be passed to the center node \cite{BiTGCF}.
```
